packages/swapper/src/swappers/RelaySwapper/endpoints.ts

```diff
@@ -0,0 +1,170 @@
+import { fromChainId } from '@shapeshiftoss/caip'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { Result } from '@sniptt/monads/build'
+import { Err } from '@sniptt/monads/build'
+import type { InterpolationOptions } from 'node-polyglot'
+
+import type {
+  CommonTradeQuoteInput,
+  EvmTransactionRequest,
+  GetTradeRateInput,
+  GetUnsignedEvmTransactionArgs,
+  SwapErrorRight,
+  SwapperApi,
+  SwapperDeps,
+  TradeQuote,
+  TradeRate,
+} from '../../types'
+import { TradeQuoteError } from '../../types'
+import { checkSafeTransactionStatus, isExecutableTradeQuote, makeSwapErrorRight } from '../../utils'
+import { relayChainMap } from './constant'
+import { getTradeQuote } from './getTradeQuote/getTradeQuote'
+import { getTradeRate } from './getTradeRate/getTradeRate'
+import { relayService } from './utils/relayService'
+import type { RelayStatus } from './utils/types'
+
+export const relayApi: SwapperApi = {
+  getTradeQuote: async (
+    input: CommonTradeQuoteInput,
+    deps: SwapperDeps,
+  ): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+    if (input.sellAmountIncludingProtocolFeesCryptoBaseUnit === '0') {
+      return Err(
+        makeSwapErrorRight({
+          message: 'sell amount too low',
+          code: TradeQuoteError.SellAmountBelowMinimum,
+        }),
+      )
+    }
+
+    const tradeQuoteResult = await getTradeQuote(input, deps, relayChainMap)
+
+    return tradeQuoteResult
+  },
+  getTradeRate: async (
+    input: GetTradeRateInput,
+    deps: SwapperDeps,
+  ): Promise<Result<TradeRate[], SwapErrorRight>> => {
+    if (input.sellAmountIncludingProtocolFeesCryptoBaseUnit === '0') {
+      return Err(
+        makeSwapErrorRight({
+          message: 'sell amount too low',
+          code: TradeQuoteError.SellAmountBelowMinimum,
+        }),
+      )
+    }
+
+    const tradeRateResult = await getTradeRate(input, deps, relayChainMap)
+
+    return tradeRateResult
+  },
+  getUnsignedEvmTransaction: async ({
+    chainId,
+    from,
+    stepIndex,
+    tradeQuote,
+  }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const { to, value, data, gas, maxFeePerGas, maxPriorityFeePerGas } =
+      tradeQuote.steps[stepIndex]?.relayTransactionMetadata ?? {}
+
+    if (
+      to === undefined ||
+      value === undefined ||
+      data === undefined ||
+      gas === undefined ||
+      maxFeePerGas === undefined ||
+      maxPriorityFeePerGas === undefined
+    ) {
+      const undefinedRequiredValues = [
+        to,
+        value,
+        data,
+        gas,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+      ].filter(value => value === undefined)
+
+      throw Error('undefined required values in swap step', {
+        cause: {
+          undefinedRequiredValues,
+        },
+      })
+    }
+
+    return await Promise.resolve({
```

Comment:
?

Reply:
`SwapperApi` only supports async functions which make sense 99% of the time, I was doing this to trick it but updating the `SwapperApi` type fixed it (in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03) )

Reply:
`return Promise.resolve()` is fine but we should never `return await somePromise`.

Reply:
Please revert the above changes to keep things tidy, `Promise<EvmTransactionRequest> | EvmTransactionRequest` seems like a bug waiting to happen.

Reply:
It has already been removed in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03)

Reply:
FYI I changed a few bits in [29fc44d](https://github.com/shapeshift/web/pull/9188/commits/29fc44d8ce158086f2cbec2c3c7ee0dd44302dac)